bin/exec.go

```diff
@@ -87,73 +87,45 @@ func main() {
 			break
 		}
 
-		var perExperiment utils.ExperimentDetails
-
 		log.Infoln("Getting the Default ENV Variables")
 
-		// Get thee Deafult ENV's from ChaosExperiment
-		perExperiment.Env = utils.GetEnvFromExperiment(engineDetails.AppNamespace, engineDetails.Experiments[i], engineDetails.Config)
+		// Get thee Default ENV's from ChaosExperiment
+		//perExperiment.Env = utils.GetEnvFromExperiment(engineDetails, engineDetails.Experiments[i])
+		perExperiment.SetDefaultEnv(clients)
 
 		log.Info("Printing the Default Variables", perExperiment.Env)
 
 		// Get the ConfigMaps for patching them in the job creation
 		log.Infoln("Find the configMaps in the chaosExperiments")
 
-		configMapExist, configMaps := utils.CheckConfigMaps(engineDetails, config, engineDetails.Experiments[i])
-
-		var validatedConfigMaps []v1alpha1.ConfigMap
-		var errorsListForConfigMaps []error
-		if configMapExist == true {
-			log.Infoln("Config Maps Found")
-			//fetch details and apply those config maps needed
-			// to be used in the job creation
-			// first convert the format of ConfigMap's Data to map[string]string
-			// & then use the kube-builder to build config maps
-			validatedConfigMaps, errorsListForConfigMaps = utils.ValidateConfigMaps(configMaps, engineDetails, clients)
-
-			if errorsListForConfigMaps != nil {
-				log.Errorf("Printing Errors, found while Validating ConfigMaps : %v, Will abort the Experiment Execution", errorsListForConfigMaps)
-				continue
-			}
-		} else {
-			log.Infoln("Unable to find ConfigMaps")
-		}
+		perExperiment.SetConfigMaps(clients)
 
-		secretsExist, secrets := utils.CheckSecrets(engineDetails, config, engineDetails.Experiments[i])
-
-		var validatedSecrets []v1alpha1.Secret
-		var errorsListForSecrets []error
-		if secretsExist == true {
-			log.Infoln("Secrets Found")
-			//fetch details and apply those config maps needed
-			// to be used in the job creation
-			// first convert the format of ConfigMap's Data to map[string]string
-			// & then use the kube-builder to build config maps
-			validatedSecrets, errorsListForSecrets = utils.ValidateSecrets(secrets, engineDetails, clients)
-
-			//log.Infoln("Printing VolumeMounts : ", volumeMounts)
-			if errorsListForSecrets != nil {
-				log.Errorf("Printing Errors, found while Validating Secrets : %v, Will abort the Experiment Execution", errorsListForSecrets)
-				continue
-			}
-		} else {
-			log.Infoln("Unable to find Secrets")
+		err := perExperiment.ValidateConfigMaps(clients)
+		if err != nil {
+			log.Infof("Aborting Execution")
```

Comment:
Can you make the log message more understandable

Reply:
Logged it before, in the function itself, from where the error is returning

Reply:
Can you add some more text in the log message which will explain from where this log message is showing and why

Reply:
THis will be a sequentially logging, user will get context.

Reply:
okay! As per discussed might get refactored in verbosity logging